include/FixedPointConverter.h

```diff
@@ -444,13 +468,83 @@ namespace ChimeraTK {
   };
 
   /********************************************************************************************************************/
-  template<>
-  [[nodiscard]] uint32_t FixedPointConverter::toRaw<std::string>(std::string cookedValue) const;
+  /* As the class is templated now, move implementation from the source file*/
+  /********************************************************************************************************************/
+  template<typename RawType>
+  const int FixedPointConverter<RawType>::zero = 0;
+
+  template<typename RawType>
+  FixedPointConverter<RawType>::FixedPointConverter(
+      std::string variableName, unsigned int nBits, int fractionalBits, bool isSignedFlag)
+  : _variableName(std::move(variableName)), _nBits(nBits), _fractionalBits(fractionalBits), _isSigned(isSignedFlag),
+    _fractionalBitsCoefficient(pow(2., -fractionalBits)), _inverseFractionalBitsCoefficient(pow(2., fractionalBits)) {
+    reconfigure(nBits, fractionalBits, isSignedFlag);
+  }
+
+  template<typename RawType>
+  void FixedPointConverter<RawType>::reconfigure(unsigned int nBits, int fractionalBits, bool isSignedFlag) {
+    _nBits = nBits;
+    _fractionalBits = fractionalBits;
+    _isSigned = isSignedFlag;
+    _fractionalBitsCoefficient = pow(2., -fractionalBits);
+    _inverseFractionalBitsCoefficient = pow(2., fractionalBits);
+
+    _bitShiftMaskSigned = 0; // Fixme unused variable.
+    // some sanity checks
+    if(nBits > 32) {
+      std::stringstream errorMessage;
+      errorMessage << "The number of bits must be <= 32, but is " << nBits;
+      throw ChimeraTK::logic_error(errorMessage.str());
+    }
+
+    // For floating-point types: check if number of fractional bits are complying
+    // with the dynamic range Note: positive fractional bits give us smaller
+    // numbers and thus correspond to negative exponents!
+    if((fractionalBits > -std::numeric_limits<double>::min_exponent - static_cast<int>(nBits)) ||
+        (fractionalBits < -std::numeric_limits<double>::max_exponent + static_cast<int>(nBits))) {
+      std::stringstream errorMessage;
+      errorMessage << "The number of fractional bits exceeds the dynamic"
+                   << " range of a double.";
+      throw ChimeraTK::logic_error(errorMessage.str());
+    }
+
+    // compute mask for the signed bit
+    // keep the mask at 0 if unsigned to simplify further calculations
+    if(nBits > 0) {
+      // NOLINTNEXTLINE(hicpp-signed-bitwise)
+      _signBitMask = (_isSigned ? 1 << (nBits - 1) : 0x0); // the highest valid bit is the sign
+    }
+    else {
+      _signBitMask = 0;
+    }
+
+    // compute masks of used and unused bits
+    // by using 1L (64 bit) this also works for 32 bits, which needs 33 bits during the calculation
+    // NOLINTNEXTLINE(hicpp-signed-bitwise)
+    _usedBitsMask = static_cast<RawType>((1L << nBits) - 1L);
```

Comment:
This hack is gonna fail for 64 bit integers because we would need 65 a bit container.

Reply:
Yep, that looks good.